JavaScript/JavaScriptSDK.Tests/CheckinTests/Sender.tests.ts

```diff
@@ -655,12 +666,497 @@ class SenderTests extends TestClass {
+                });
+
+                XDomainRequest = <any>(() => {
+                    var xdr = new this.xhr;
+                    xdr.onload = xdr.onreadystatechange;
+                    xdr.responseText = 206;
+                    return xdr;
+                });
+
+                this.validatePartialSuccess_Retryable();
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: XDomain - can disable partial response handling",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                this.validatePartialSuccess_disabled();
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: ParseResponse - invalid number of errors",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                var sender = this.getSender();
+                Assert.ok(sender, "sender was constructed");
+
+                // too many errors
+                var response = '{ "itemsReceived": 2, "itemsAccepted": 1, "errors": [{ "index": 0, "statusCode": 408, "message": "error" }, { "index": 2, "statusCode": 429, "message": "error" }] }';
+                var result = <Microsoft.ApplicationInsights.IBackendResponse>(<any>sender)._parseResponse(response);
+
+                Assert.ok(!result, "Parse should fail when there are too many errors");
+
+                // no errors
+                response = '{ "itemsReceived": 2, "itemsAccepted": 1, "errors": [] }';
+                result = <Microsoft.ApplicationInsights.IBackendResponse>(<any>sender)._parseResponse(response);
+
+                Assert.ok(!result, "Parse should fail - there should be one error");
+
+                // no warnings
+                this.logAsserts(0);
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: ParseResponse - invalid number of accepted items",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                var sender = this.getSender();
+                Assert.ok(sender, "sender was constructed");
+
+                // too many items accepted
+                var response = '{ "itemsReceived": 1, "itemsAccepted": 2, "errors": [] }';
+                var result = <Microsoft.ApplicationInsights.IBackendResponse>(<any>sender)._parseResponse(response);
+
+                Assert.ok(!result, "Parse should fail - there are too itemsAccepted > itemsReceived");
```

Comment:
1) Yes and no. In theory it shouldn't happen, but I wanted to test if our code doesn't throw when backend returns inconsistent response. 

2) "too many". I improved the comment to explain that there were two errors instead of one.